config/versions/register_test.go

```diff
@@ -0,0 +1,49 @@
+package versions_test
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/thrasher-corp/gocryptotrader/config/versions"
+	testutils "github.com/thrasher-corp/gocryptotrader/internal/testing/utils"
+)
+
+func TestVersionsRegistered(t *testing.T) {
+	t.Parallel()
+
+	r, err := testutils.RootPathFromCWD()
+	require.NoError(t, err)
+
+	versionsDir := filepath.Join(r, "config", "versions")
+	_, err = os.Stat(versionsDir)
+	require.NoError(t, err, "config/versions must exist")
+
+	err = filepath.WalkDir(versionsDir, func(p string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if !d.IsDir() || p == versionsDir {
+			return nil
+		}
+		verStr := filepath.Base(p)
+		verMatch := regexp.MustCompile(`v(\d+)`).FindStringSubmatch(verStr)
```

Comment:
suggestion: if somebody starts doing v1.1 folder we might be able to screen that.

Reply:
Sorry, what does "might be able to screen that" mean?
You mean: If someone uses an incorrect version number format, we might be able to error on it?

Reply:
Yeah exactly like somebody might introduce a minor version convention, it's pretty low on the probability though.

Reply:
This is where I like tests that enforce design. Gk's test ensures noone can release v13.37 and make people sad

Reply:
No I didn't know what I was talking about with this, I don't know how I tested it but when I checked again properly it all errored correctly. 🤷 I obviously am being a silly billy and will resolve this.